pkg/frontend/querymiddleware/instrumentation.go

```diff
@@ -32,8 +31,10 @@ func newInstrumentMiddleware(name string, metrics *instrumentMiddlewareMetrics,
 		return HandlerFunc(func(ctx context.Context, req Request) (Response, error) {
 			var resp Response
 			err := instrument.CollectedRequest(ctx, name, durationCol, instrument.ErrorCode, func(ctx context.Context) error {
-				sp := spanlogger.FromContext(ctx, logger)
-				req.LogToSpan(sp.Span)
+				sp := opentracing.SpanFromContext(ctx)
```

Comment:
nit: I don't understand why we prefer spanlogger or using opentracing directly here and elsewhere.

Reply:
`SpanLogger` is-a `Span` and is-a `Logger`.
It's good for places where you want to write someting like `level.Warn(logger).Log("msg", ...)` and have that visible in both the logfile and in a span.

In this particular file, it was only used as a `Span` so the extra wrapper cost could be eliminated.